src/circuit/ecc.rs

```diff
@@ -48,13 +72,19 @@ pub trait EccInstruction<C: CurveAffine> {
     ) -> Result<AssignedPoint<C>, Error>;
     fn add(&self, region: &mut Region<'_, C::ScalarExt>, p0: AssignedPoint<C>, p1: AssignedPoint<C>, offset: &mut usize) -> Result<AssignedPoint<C>, Error>;
     fn double(&self, region: &mut Region<'_, C::ScalarExt>, p: AssignedPoint<C>, offset: &mut usize) -> Result<AssignedPoint<C>, Error>;
-    fn mul_var(&self, region: &mut Region<'_, C::ScalarExt>, p: AssignedPoint<C>, e: C::ScalarExt, offset: &mut usize) -> Result<AssignedPoint<C>, Error>;
-    fn mul_fix(&self, region: &mut Region<'_, C::ScalarExt>, p: C, e: C::ScalarExt, offset: &mut usize) -> Result<AssignedPoint<C>, Error>;
+    fn mul_var(
+        &self,
+        region: &mut Region<'_, C::ScalarExt>,
+        p: AssignedPoint<C>,
+        e: AssignedInteger<C::ScalarExt>,
```

Comment:
Scalar field element is not big integer rather it is native to the field we are working.

We emulate base field `Fq` of the curve in its scalar field `Fr` (which we also call native). So multiplier is already native element.

Reply:
The original trait can only support `mul` whose scalar is a constant rather than an `Assigned` object which is needed by ECDSA verification scheme.

Reply:
Yes that's right I understand reason of the change but still exponent should be `AssignedValue` rather than `AssignedInteger`. Since `AssingedValue`, represents scalar field and `AssignedInteger` represents base field

Reply:
I get the point